packages/contracts-bedrock/src/Safe/LivenessModule.sol

```diff
@@ -0,0 +1,148 @@
+            console.log("removing one owner. numOwners: %s, thresholdAfter: %s", numOwners, thresholdAfter);
+            safe.execTransactionFromModule({
+                to: address(safe),
+                value: 0,
+                data: abi.encodeCall(
+                    // Call the Safe to remove the owner
+                    OwnerManager.removeOwner,
+                    (getPrevOwner(owner, owners), owner, thresholdAfter)
+                    ),
+                operation: Enum.Operation.Call
+            });
+        } else {
+            console.log("removing all owners. numOwnersAfter: %s", numOwners);
+            // The number of owners is dangerously low, so we wish to transfer the ownership of this Safe to a new
+            // to the fallback owner.
+
+            // The threshold will be 1 because we are removing all owners except the fallback owner
+            thresholdAfter = 1;
+
+            // Remove owners one at a time starting from the last owner.
+            // Since we're removing them in order, the ordering will remain constant,
+            //  and we shouldn't need to query the list of owners again.
+            for (uint256 i = owners.length - 1; i >= 0; i--) {
+                address currentOwner = owners[i];
+                address prevOwner = getPrevOwner(currentOwner, owners);
+                if (currentOwner != address(this)) {
+                    safe.execTransactionFromModule({
+                        to: address(safe),
+                        value: 0,
+                        data: abi.encodeCall(
+                            // Call the Safe to remove the owner
+                            OwnerManager.removeOwner,
+                            (prevOwner, currentOwner, 1)
+                            ),
+                        operation: Enum.Operation.Call
+                    });
+                }
+            }
+
+            // Add the fallback owner as the sole owner of the Safe
+            safe.execTransactionFromModule({
+                to: address(safe),
+                value: 0,
+                data: abi.encodeCall(OwnerManager.addOwnerWithThreshold, (fallbackOwner, 1)),
+                operation: Enum.Operation.Call
+            });
+        }
+        _verifyFinalState();
+    }
+
+    /// @notice A FREI-PI invariant check enforcing requirements on number of owners and threshold.
+    function _verifyFinalState() internal view {
+        address[] memory owners = safe.getOwners();
+        uint256 numOwners = owners.length;
+        require(
+            (numOwners == 1 && owners[0] == fallbackOwner) || (numOwners >= minOwners),
+            "LivenessModule: Safe must have at least 1 owner or minOwners"
+        );
+
+        // Check that the threshold is correct
+        uint256 threshold = safe.getThreshold();
+        require(
+            threshold == get75PercentThreshold(numOwners) || (numOwners == 1 && threshold == 1),
```

Comment:
I think the `|| (numOwners == 1 && threshold == 1)` check is redundant? If we have 1 owner, we still need `threshold == get75PercentThreshold(numOwners) == 1`

Reply:
Good point. I broke the `get75PercentThreshold` check out on its own, so it always runs. 

The consideration as to whether the threshold is correct is now separate from whether the number of owners is correct.

https://github.com/ethereum-optimism/optimism/blob/jm/safe-liveness/packages/contracts-bedrock/src/Safe/LivenessModule.sol#L137-L148